doc/book/cookbook/using-a-base-path.md

```diff
@@ -0,0 +1,176 @@
+this environment?
+
+## .htaccess in the application root.
+
+If you are using Apache, your first step is to add an `.htaccess` file to your
+application root, with directives for rewriting to the `public/` directory:
+
+```ApacheConf
+RewriteEngine On
+RewriteRule (.*) ./public/$1
+```
+
+> ### Using other web servers
+>
+> If you are using a web-server other than Apache, and know how to do a similar
+> rewrite, we'd love to know! Please submit ideas/instructions to
+> [our issue tracker](https://github.com/zendframework/zend-expressive/issues)!
+
+## Use middleware to rewrite the path
+
+The above step ensures that clients can hit the website. Now we need to ensure
+that the application can route to middleware!
+
+To do this, we will add pre_routing pipeline middleware to intercept the
+request, and rewrite the URL accordingly.
+
+At the time of writing, we have two suggestions:
+
+- [los/basepath](https://github.com/Lansoweb/basepath) provides the basic
+  mechanics of rewriting the URL, and has a stable release.
+- [mtymek/blast-base-url](https://github.com/mtymek/blast-base-url) provides the
+  URL rewriting mechanics, as well as utilities for generating URIs that retain
+  the base path, but does not have a stable release yet.
+
+### los/basepath
+
+To use `los/basepath`, install it via Composer, copy the configuration files to
+your application, and then edit the configuration.
+
+To install and copy the configuration:
+
+```bash
+$ composer require los/basepath
+$ cp vendor/los/basepath/config/los-basepath.global.php.dist config/autoload/los-basepath.global.php
+```
+
+We recommend copying the global configuration to a local configuration file as
+well; this allows you to have the production settings in your global
+configuration, and development settings in a local configuration (which is
+excluded from git by default):
+
+```bash
+$ cp config/autoload/los-basepath.global.php config/autoload/los-basepath.local.php
+```
+
+Then edit one or both, to change the `los_basepath` settings:
+
+```php
+return [
+    'los_basepath' => '<base path here>',
+    /* ... */
+];
+```
+
+The base path should be the portion of the web root leading up to the
+`index.php` of your application. In the above example, this would be
+`/expressive`.
+
+### mtymek/blast-base-url
+
+To use `mtymek/blast-base-url`, install it via Composer, and register some
+configuration.
+
+To install it:
+
+```bash
+$ composer require "mtymek/blast-base-url:dev-master@dev"
+```
+
+To configure it, update the file `config/autoload/middleware-pipeline.global.php`,
+with the following contents:
+
+```php
+return [
+    'dependencies' => [
+        'factories' => [
+            Blast\BaseUrl\BaseUrlMiddleware::class => Blast\BaseUrl\BaseUrlMiddlewareFactory::class,
+            /* ... */
+        ],
+        /* ... */
+    ],
+    'middleware_pipeline' => [
+        'pre_routing' => [
+            [ 'middleware' => [ BaseUrlMiddleware::class ] ],
```

Comment:
it won't work if copy-pasted directly - maybe class import or FQCN should be used?